server/src/test/java/org/elasticsearch/index/mapper/IpFieldMapperTests.java

```diff
@@ -265,33 +169,29 @@ public void testNullValue() throws IOException {
         assertEquals(DocValuesType.SORTED_SET, dvField.fieldType().docValuesType());
         assertEquals(new BytesRef(InetAddressPoint.encode(InetAddresses.forString("::1"))), dvField.binaryValue());
         assertFalse(dvField.fieldType().stored());
-    }
 
-    public void testSerializeDefaults() throws Exception {
-        String mapping = Strings.toString(XContentFactory.jsonBuilder().startObject().startObject("type")
-            .startObject("properties").startObject("field").field("type", "ip").endObject().endObject()
-            .endObject().endObject());
+        mapper = createDocumentMapper(fieldMapping(b -> {
+            b.field("type", "ip");
+            b.nullField("null_value");
+        }));
 
-        DocumentMapper docMapper = parser.parse("type", new CompressedXContent(mapping));
-        IpFieldMapper mapper = (IpFieldMapper)docMapper.root().getMapper("field");
-        XContentBuilder builder = XContentFactory.jsonBuilder().startObject();
-        mapper.doXContentBody(builder, true, ToXContent.EMPTY_PARAMS);
-        String got = Strings.toString(builder.endObject());
+        doc = mapper.parse(source(b -> b.nullField("field")));
+        assertArrayEquals(new IndexableField[0], doc.rootDoc().getFields("field"));
 
-        // it would be nice to check the entire serialized default mapper, but there are
-        // a whole lot of bogus settings right now it picks up from calling super.doXContentBody...
-        assertTrue(got, got.contains("\"ignore_malformed\":false"));
-    }
+        mapper = createDocumentMapper(fieldMapping(b -> {
+            b.field("type", "ip");
+            b.field("null_value", "");
+        }));
 
-    public void testEmptyName() throws IOException {
```

Comment:
testEmptyName and testSerializeDefaults are done as part of the parent class tests.  testMeta in the parent class discovered that IpFieldMapper wasn't correctly serializing its meta param, so the test cutover is definitely worth it!